test/MUIDataTable.test.js

```diff
@@ -903,6 +902,88 @@ describe('<MUIDataTable />', function() {
     assert.equal(state.displayData.length, data.length);
   });
 
+  describe('should displayData consider filterOptions with logic', () => {
+
+    it('with one column', () => {
+
+      const customFilterColumns = columns.map(c => {
+        if (c.name === "Name") return {
+          ...c,
+          options: {
+            ...c.options,
+            filterOptions: {
+              names: ["B", "J"],
+              logic(name, filters) {
+                const firstLetter = name[0];
+                return firstLetter !== filters[0];
+              }
+            },
+          }
+        };
+        return c;
+      });
+  
+      const shallowWrapper = shallow(<MUIDataTable columns={customFilterColumns} data={data} />);
+      const table = shallowWrapper.dive();
+      const instance = table.instance();
+      instance.filterUpdate(0, 'J', 'checkbox');
```

Comment:
`filterUpdate` actually takes 4 arguments, so the 3rd should be the column name, as in `instance.filterUpdate(0, 'J', 'Names', 'checkbox');`

Reply:
ok